tests/framework/extensions/charts/ranchercisbenchmark.go

```diff
@@ -0,0 +1,198 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/api/steve/catalog/types"
+	catalogv1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/defaults"
+	kubenamespaces "github.com/rancher/rancher/tests/framework/extensions/kubeapi/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	// Namespace that rancher CIS Benchmark chart is installed in
+	RancherCisBenchmarkNamespace = "cis-operator-system"
+	// Name of the rancher CIS Benchmark chart
+	RancherCisBenchmarkName = "rancher-cis-benchmark"
+	// Name of rancher CIS Benchmark crd chart
+	RancherCisBenchmarkCRDName = "rancher-cis-benchmark-crd"
+)
+
+// InstallRancher CIS BenchmarkChart is a helper function that installs the rancher-CIS Benchmark chart.
+func InstallRancherCisBenchmarkChart(client *rancher.Client, installOptions *InstallOptions) error {
```

Comment:
Can we have this function only for installing the chart? Code from `// Cleanup registration` , can we include in another function?

Reply:
What would be the benefit of abstracting it? We need to register created resources by using cleanup registration; the same applies to cluster creation functions as well, so at the teardown, we can trigger those registered functions to the session, depending on the given config's cleanup boolean option.

I think separating that would only be beneficial if this function's cleanup registration is used that abstracted delete function and if it's used on f.e. in a delete-specific test case.

Reply:
I have been working on Rbac PR for taking an etcd snapshot, and in there, creating a separate function for taking a snapshot is all i wanted to work with. I can see a similar usecase where, I am testing only a chart install action.

Reply:
Ah, I see; in that case, can't we provide a cleanup to false and run the chart? If the flag isn't set, the session cleanup won't be triggered, and this function becomes only a chart install action as it is

Reply:
No I disagree completely we need the cleanup registration here to ensure that the chart is uninstalled correctly. We do a clean up here because there needs to be a wait and cleanup is lot more involved. I think you misunderstand the functionality of clean up Sowmya. The clean up only occurs if the session.Cleanup call is made, or if clean up is set to true in the config. If you want to avoid cleanup then you don't make that call, or set cleanup to false.

Reply:
I agree with Sowmya. If chart uninstall action is separated, it can be served for the following purpose:

- We can verify the uninstall action if the chart is already installed manually and just verifying uninstall removes all the required crds and resources.
- This is not tightly bound together with the install action

Reply:
We can abstract the uninstall action no problem, I don't have an issue with that. But the clean up registration still needs to happen regardless in this function I think that is the confusion. By default we want to be able uninstall the chart cleanly. If not we don't uninstall Does that make sense? If we don't want the uninstall action to happen we can set the flag to false.

Reply:
Spoke to @igomez06  offline. `client.Session.RegisterCleanupFunc` is tracking the resource for clean up as explained here - https://github.com/rancher/rancher/tree/release/v2.7/tests#sessions. We can leave this function be.

Reply:
thank you all for the clarifications.